elm/src/main/scala/io/scalac/elm/state/ElmMinState.scala

```diff
@@ -51,9 +51,9 @@ case class ElmMinState(storage: Map[ByteKey, TxOutput] = Map())
     val outputSum = tx.outputs.map(_.value).sum + tx.fee
     val addsUp = inputSum == outputSum
 
-    lazy val positiveOuts = tx.outputs.forall(_.value > 0)
+    val positiveOuts = tx.outputs.forall(_.value > 0)
 
-    lazy val signed = tx.inputs.forall { in =>
+    val signed = tx.inputs.forall { in =>
```

Comment:
The purpose of `lazy` is that we don't want to check the second and third conditions if the first one fails. I can't imagine how this could affect the final result. Maybe there was something wrong your debug method? I'm going to try with `lazy` in place.